frontend/src/metabase/components/Button.jsx

```diff
@@ -0,0 +1,35 @@
+import React, { Component, PropTypes } from "react";
+
+import Icon from "metabase/components/Icon.jsx";
+
+import cx from "classnames";
+
+import _ from "underscore";
+
+const BUTTON_VARIANTS = [
+    "small",
+    "medium",
+    "large",
+    "primary",
+    "warning",
+    "cancel",
+    "purple",
+    "borderless"
+];
+
+const Button = ({ className, icon, children, ...props }) => {
+    let variantClasses = BUTTON_VARIANTS.filter(variant => props[variant]).map(variant => "Button--" + variant);
```

Comment:
So is the general idea of the usage here that you'd do something like `<Button large cancel {...props} />` to get a large cancel button? I think overall I like that approach but it does make me really want to have pretty strict proptypes so that usage is a bit more clear. I _think_ the various button classes should all be isolated enough that you won't get weird effects when combining them but we should double check.